landlab/components/landslides/landslide_probability.py

```diff
@@ -0,0 +1,628 @@
+#!/usr/env/python
+
+"""Landlab component that simulates landslide probability of failure as well
+as mean relative wetness and probability of saturation.
+
+Relative wetness and factor-of-safety are based on the infinite slope
+stability model driven by topographic and soils inputs and recharge provided
+by user as inputs to the component. For each node, component simulates mean
+relative wetness as well as the probability of saturation based on Monte Carlo
+simulation of relative wetness where the probability is the number of
+iterations with relative wetness >= 1.0 divided by the number of iterations.
+Probability of failure for each node is also simulated in the Monte Carlo
+simulation as the number of iterations with factor-of-safety <= 1.0
+divided by the number of iterations.
+
+.. codeauthor:: R.Strauch, E.Istanbulluoglu, & S.S.Nudurupati
+University of Washington
+
+Ref 1: Strauch et. al. 2017, 'A hydro-climatological approach to predicting
+regional landslide probability using Landlab, Earth Surface Dynamics, In prep.
+
+Ref 2: 'The Landlab LandslideProbability Component User Manual'
+
+Created on Thu Aug 20, 2015
+Last edit June 7, 2017
+"""
+
+# %% Import Libraries
+from landlab import Component
+from landlab.utils.decorators import use_file_name_or_kwds
+import numpy as np
+import scipy.constants
+from scipy import interpolate
+from statsmodels.distributions.empirical_distribution import ECDF
+import copy
+
+# %% Instantiate Object
+
+
+class LandslideProbability(Component):
+    """
+    Landlab component designed to calculate probability of failure at
+    each grid node based on the infinite slope stability model
+    stability index (Factor of Safety).
+
+    The driving force for failure is provided by the user in the form of
+    groundwater recharge; four options for providing recharge are supported.
+    The model uses topographic and soil characteristics provided as input
+    by the user.
+
+    The main method of the LandslideProbability class is
+    calculate_landslide_probability(), which calculates the mean soil relative
+    wetness, probability of soil saturation, and probability of failure at
+    each node based on a Monte Carlo simulation.
+
```

Comment:
Hey Katy, we (Ronda, me and Erkan) decided that this is not a good place to explain what 'HSD' or 'MD' is in detail. We came to the conclusion that one should refer to the paper and the user manual (especially to address this issue) if the user is inclined towards using this optional input method. To use this input, the user has to absolutely refer to the references mentioned, to even make sense of the inputs/outputs. 
However, I have made that clear now in the description.